gax/src/main/java/com/google/api/gax/rpc/ApiClientHeaderProvider.java

```diff
@@ -29,188 +29,175 @@
 
-    private Builder() {
-      generatorName = DEFAULT_GENERATOR_NAME;
-      generatorVersion = DEFAULT_VERSION;
+      resourceHeaderKey = getDefaultResourceHeaderKey();
+      resourceToken = null;
     }
 
-    private Builder(ApiClientHeaderProvider provider) {
-      this.apiClientHeaderLineKey = provider.apiClientHeaderLineKey;
-      this.clientLibName = provider.clientLibName;
-      this.clientLibVersion = provider.clientLibVersion;
-      this.generatorName = provider.generatorName;
-      this.generatorVersion = provider.generatorVersion;
-      this.apiClientHeaderLineData = provider.apiClientHeaderLineData;
-      this.googleCloudResourcePrefix = provider.googleCloudResourcePrefix;
+    public String getApiClientHeaderKey() {
+      return apiClientHeaderKey;
     }
 
-    public Builder setApiClientHeaderLineKey(String key) {
-      this.apiClientHeaderLineKey = key;
+    public Builder setApiClientHeaderKey(String apiClientHeaderKey) {
+      this.apiClientHeaderKey = apiClientHeaderKey;
       return this;
     }
 
-    /** Sets the client library name and version for the custom header. */
-    public Builder setClientLibHeader(String name, String version) {
-      this.clientLibName = name;
-      this.clientLibVersion = version;
+    public String getJvmToken() {
+      return jvmToken;
+    }
+
+    public Builder setJvmToken(String version) {
+      this.jvmToken = constructToken("gl-java", version);
       return this;
     }
 
-    /** Sets the generator name and version for the custom header. */
-    public Builder setGeneratorHeader(String name, String version) {
-      this.generatorName = name;
-      this.generatorVersion = version;
+    public String getClientLibToken() {
+      return clientLibToken;
+    }
+
+    public Builder setClientLibToken(String name, String version) {
+      this.clientLibToken = constructToken(name, version);
       return this;
     }
 
-    /**
-     * Adds a new piece of data to the end of the x-goog-api-client header key. Needs to be in the
-     * format "key/value" (without quotes).
-     */
-    public Builder addApiClientHeaderLineData(List<String> data) {
-      for (String datum : data) {
-        Preconditions.checkArgument(datum.contains("/"));
-      }
-      apiClientHeaderLineData.addAll(data);
+    public String getGeneratedLibToken() {
+      return generatedLibToken;
+    }
+
+    public Builder setGeneratedLibToken(String name, String version) {
+      this.generatedLibToken = constructToken(name, version);
       return this;
     }
 
-    /** Sets the google-cloud-resource-prefix header. */
-    @BetaApi("This API and its semantics are likely to change in the future.")
-    public Builder setGoogleCloudResourcePrefix(String resourcePrefix) {
-      this.googleCloudResourcePrefix = resourcePrefix;
+    public String getGeneratedRuntimeToken() {
+      return generatedRuntimeToken;
+    }
+
+    public Builder setGeneratedRuntimeToken(String version) {
```

Comment:
GAX isn't a "generated runtime", it's a runtime used by the generated code. I think "client runtime" would be a better generic name. But, since it's hardcoded to gax, I might prefer just "setGaxToken". Or given my other comment about the confusingness of using "token" when only the value is provided, "setGaxValue".